src/components/Toasts/ToastsProvider.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { createContext, useState } from 'react';
+import PropTypes from 'prop-types';
+
+export const ToastsContext = createContext();
+
+const ToastsProvider = ({ children }) => {
```

Comment:
[clarification] This is largely copied from frontend-app-admin-portal, correct? I'm curious if you think the `ToastProvider` approach is needed versus just rendering individual `Toast` components throughout the app. Another downside of the `addToast` is that (afaik) it can really only handle string messages. But technically, `Toasts` can be customized with add buttons/links/etc. so it's not as flexible as it could be as an abstraction layer.

I believe there are already some `Toasts` in frontend-app-admin-portal that also don't make use of the `ToastProvider`, rather just rendering them as standalone components as needed. Perhaps it's an abstraction layer that's not really needed after all?

I'm curious to get your two cents as well on the `ToastProvider` vs. rendering individual `Toasts` 😃

Reply:
Great points! I agree `ToastProvider` abstracting away the need to add `useState`/JSX to every component is a big win. Another downside of this implementation (IIRC) is that the animation (e.g., fade in) doesn't seem to be applied like it does when rendering standalone `Toast` components.

Is this something you're seeing here as well?

Reply:
I see what you mean. This can totally be solved, the reason it happens is that we only render the `Toast` component when we have one. This breaks the animation because it relies on the toast component being in the DOM before we make a state change to `show`.

I can go into detail on how we can solve this. I propose that maybe we make this fix on a separate piece of work where maybe we add this toast functionality to `frontend-enterprise` npm package to be reused across other apps.

Reply:
> I see what you mean. This can totally be solved, the reason it happens is that we only render the `Toast` component when we have one. This breaks the animation because it relies on the toast component being in the DOM before we make a state change to `show`.
> 
> I can go into detail on how we can solve this. I propose that maybe we make this fix as a separate piece of work where maybe we add this toast functionality to `frontend-enterprise` npm package to be reused across other apps.

@manny-m Yeah, it should be a relatively straightforward fix to have the toast always being in the DOM. I agree on deferring this change as part of the tech debt effort to consolidate to a shared component; the animation is not worth the scope creep right now :) Do you want to file a ticket these 2 things?